### PR TITLE
docs: update clustermesh troubleshooting guide

### PR DESCRIPTION
**Pull request by giorio94**
This PR updates some outdated pieces of information in the clustermesh troubleshooting guide, in particular about the TLS certificates.

```release-note
Update the clustermesh troubleshooting guide
```

### CODE REVIEW

File: Documentation/operations/troubleshooting_clustermesh.rst [hunk 1]
```diff
@@ -76,23 +80,23 @@ Manual Verification of Setup
...
 
-    * The Kubernetes secret ``cilium-clustermesh`` is imported correctly.
+    * The ``cilium-clustermesh`` Kubernetes secret is present and correctly
+      mounted by the Cilium agent pods.
 
-    * The secret contains a file for each remote cluster with the filename
-      matching the name of the remote cluster.
+    * The secret contains a file for each remote cluster with the filename matching
+      the name of the remote cluster (as provided by the ``--cluster-name`` argument
+      or the ``cluster-name`` ConfigMap option).
```
Comment (learnitall) [inline Documentation/operations/troubleshooting_clustermesh.rst]: I think we can remove the parentheticals here.

```suggestion
      the name of the remote cluster as provided by the ``--cluster-name`` argument
      or the ``cluster-name`` ConfigMap option.
```

**Review by thorn3r — APPROVED** [review]

File: Documentation/operations/troubleshooting_clustermesh.rst [hunk 2]
```diff
@@ -50,18 +49,23 @@ Manual Verification of Setup
...
+      containers in the clustermesh-apiserver deployment, to authenticate against the
+      sidecar etcd instance. Not applicable if an external etcd cluster is used.
 
-    * cilium-ca, which is CA used to generate the above two certs.
+    * ``clustermesh-apiserver-remote-cert``, which is used by Cilium agents, and
+      optionally the kvstoremesh container in the clustermesh-apiserver deployment,
+      to authenticate against remote etcd instances (either internal or external).
 
-    If any of above secrets are not configured correctly, there will be potential error message like below::
+    If any of above secrets is not configured correctly, there will be potential error message like below::
```
Comment (learnitall) [inline Documentation/operations/troubleshooting_clustermesh.rst]: The words 'above' and 'below' give translators some trouble, can we change to 'prior' and 'following'?
```suggestion
    If any of the prior secrets is not configured correctly, there will be a potential error message like the following::
```

File: Documentation/operations/troubleshooting_clustermesh.rst [hunk 3]
```diff
@@ -76,23 +80,23 @@ Manual Verification of Setup
...
+      the name of the remote cluster (as provided by the ``--cluster-name`` argument
+      or the ``cluster-name`` ConfigMap option).
 
-    * The contents of the file in the secret is a valid etcd configuration
-      consisting of the IP to reach the remote etcd as well as the required
-      certificates to connect to that etcd.
+    * Each file named after a remote cluster contains a valid etcd configuration
+      consisting of the endpoints to reach the remote etcd cluster, and the path
+      of the certificate and private key to authenticate against that etcd cluster.
+      Additional files may represent the certificate and private key themselves.
```
Comment (learnitall) [inline Documentation/operations/troubleshooting_clustermesh.rst]: This sentence is a little unclear, I'm not sure what it means. I tried to add some context:

```suggestion
      Additional files may be included in the secret to provide the certificate and private key themselves.
```

File: Documentation/operations/troubleshooting_clustermesh.rst [hunk 4]
```diff
@@ -76,23 +80,23 @@ Manual Verification of Setup
...
-      required SSL certificates and keys. The filenames must match the cluster
-      names as provided by the ``--cluster-name`` argument or ``cluster-name``
-      ConfigMap option. If the directory is empty or incomplete, regenerate the
-      secret again and ensure that the secret is correctly mounted into the
-      DaemonSet.
+    * The ``/var/lib/cilium/clustermesh`` directory inside any of the Cilium agent
+      pods contains the files mounted from the ``cilium-clustermesh`` secret.
+      You can use
+      ``kubectl exec -ti -n kube-system ds/cilium -c cilium-agent -- ls /var/lib/cilium/clustermesh``
+      to list the files therein present.
```
Comment (learnitall) [inline Documentation/operations/troubleshooting_clustermesh.rst]: I think `present` and `therein` are redundant. Can you pick one and remove the other?

**Review by learnitall — CHANGES_REQUESTED** [review]
Thank you for doing this. I have a couple of suggestions and then LGTM.

**Comment by giorio94** [comment]
> Thank you for doing this. I have a couple of suggestions and then LGTM.

Suggestions applied, thanks!

**Review by learnitall — APPROVED** [review]
Thanks!

**Pull request merged**